### PR TITLE
MINIFICPP-1261 - Mark ScopeGuard class deprecated

### PR DESCRIPTION
**Pull request by hunyadi-dev**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/opc/src/opc.cpp [hunk 1]
```diff
@@ -302,7 +303,7 @@ void Client::traverse(UA_NodeId nodeId, std::function<nodeFoundCallBackFunc> cb,
 
   UA_BrowseResponse bResp = UA_Client_Service_browse(client_, bReq);
 
-  utils::ScopeGuard guard([&bResp]() {
+  gsl::finally guard([&bResp]() {
```
Comment (szaszm) [inline extensions/opc/src/opc.cpp:306]: `gsl::finally` is a function, not a type. 
```suggestion
  const auto guard = gsl::finally([&bResp] {
```

Comment (hunyadi-dev) [reply extensions/opc/src/opc.cpp]: PR updated with the correct signature.

File: extensions/windows-event-log/ConsumeWindowsEventLog.cpp [hunk 2]
```diff
@@ -340,7 +341,7 @@ void ConsumeWindowsEventLog::onTrigger(const std::shared_ptr<core::ProcessContex
 
   size_t eventCount = 0;
   const TimeDiff timeDiff;
-  utils::ScopeGuard timeGuard([&]() {
+  gsl::finally timeGuard([&]() {
```
Comment (adamdebreceni) [inline extensions/windows-event-log/ConsumeWindowsEventLog.cpp:344]: same as [this](https://github.com/apache/nifi-minifi-cpp/pull/849#discussion_r460123115)

Comment (hunyadi-dev) [reply extensions/windows-event-log/ConsumeWindowsEventLog.cpp:344]: Missed this one. :S

File: extensions/windows-event-log/ConsumeWindowsEventLog.cpp [hunk 3]
```diff
@@ -353,7 +354,7 @@ void ConsumeWindowsEventLog::onTrigger(const std::shared_ptr<core::ProcessContex
     context->yield();
     return;
   }
-  const utils::ScopeGuard guard_hEventResults([hEventResults]() { EvtClose(hEventResults); });
+  const gsl::finally guard_hEventResults([hEventResults]() { EvtClose(hEventResults); });
```
Comment (adamdebreceni) [inline extensions/windows-event-log/ConsumeWindowsEventLog.cpp:357]: same as [this](https://github.com/apache/nifi-minifi-cpp/pull/849#discussion_r460123115)

File: extensions/windows-event-log/ConsumeWindowsEventLog.cpp [hunk 4]
```diff
@@ -385,7 +386,7 @@ void ConsumeWindowsEventLog::onTrigger(const std::shared_ptr<core::ProcessContex
       }
       break;
     }
-    const utils::ScopeGuard guard_hEvent([hEvent]() { EvtClose(hEvent); });
+    const gsl::finally guard_hEvent([hEvent]() { EvtClose(hEvent); });
```
Comment (adamdebreceni) [inline extensions/windows-event-log/ConsumeWindowsEventLog.cpp:389]: here as well

**Pull request closed without merging**